get_local_image_transform_command.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright 2020–2021 CEA
+#
+# Author: Joël Chavas <joel.chavas@cea.fr>
+#
+# Licensed under the Apache Licence, Version 2.0 (the "Licence");
+# you may not use this file except in compliance with the Licence.
+# You may obtain a copy of the Licence at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the Licence is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the Licence for the specific language governing permissions and
+# limitations under the Licence.
+
+import sys
+import os
+
+import requests
+import argparse
+
+"""
+Source and target spaces can be one of the 4 spaces:
+"MNI 152 ICBM 2009c Nonlinear Asymmetric"
+"MNI Colin 27"
+"Big Brain (Histology)"
+"Infant Atlas"
+"""
+SOURCE_SPACE_DEFAULT = "MNI 152 ICBM 2009c Nonlinear Asymmetric"
+TARGET_SPACE_DEFAULT = "Big Brain (Histology)"
+
+
+SERVER_ADDRESS_DEFAULT = "http://0.0.0.0:8080"
+# SERVER_ADDRESS_DEFAULT = "http://172.17.0.1:8080"
+# SERVER_ADDRESS_DEFAULT = "https://hbp-spatial-backend.apps-dev.hbp.eu"
+INSTANCE_PATH = "/instance"
+
+
+def is_transform(str_in):
+    """Determines if str_in is the name of a transformation file
+    """
+    return '.ima' in str_in \
+        or '.trm' in str_in
+
+
+def add_path_to_transform(str_in):
+    """Adds as prefix INSTANCE_PATH to str_in if str_in is a transformaton file
+    """
+    if is_transform(str_in):
+        return f'{INSTANCE_PATH}/{str_in}'
+    else:
+        return str_in
+
+
+def get_command(server_address, source_space, target_space):
+    """Gets the image transform commmand from the server
+    """
+    request_string = \
+        f'{server_address}/v1/'\
+        f'get-image-transform-command?source_space={source_space}'\
+        f'&target_space={target_space}&input_coords=auto'
+    return requests.get(request_string)
+
+
+def format_command(command_from_server, input_file, output_file, extra):
+    """Formats the command to a string that can be run on a bash shell
+    """
+    command_as_list = command_from_server.json()['transform_command']
+    command_as_list = [add_path_to_transform(x) for x in command_as_list]
```

Comment:
Just for extra safety, I would apply `shlex.quote` to all the commandline arguments before printing then on the command-line. Especially since you use `os.system` to run the command, not quoting the arguments is a security risk (arbitrary code could be run locally if the server returns a malicious command-line).

Reply:
Still, the safest thing would be to avoid `os.system` entirely, and use `subprocess.call` instead.
Speaking of safety, you should probably verify that the command-line actually begins with `AimsApplyTransform` and nothing else (like `rm -rf /`...)